### PR TITLE
CSHARP-5399: Implement API compatibility validation EG script

### PR DESCRIPTION
**Pull request by sanych-sun**
Examples of changes that tool see as breaking change:
1) New optional parameter of existing method: https://spruce.mongodb.com/version/67258864067cc90007cafd41/
2) New members in interface: https://spruce.mongodb.com/version/672923980732b50007f95635

Please pay attention to Files tab of the failed [validate-packages] task.

Also there is ability to supply the suppression files, we will add it as we will run into any compatibility change that we do want to suppress.

### CODE REVIEW

File: evergreen/validate-apicompat.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+fi
+
+echo "Installing package validation tool."
+dotnet new tool-manifest --force
+dotnet tool install Microsoft.DotNet.ApiCompat.Tool --version 8.0.* --local
+
+EXIT_CODE=0
+mkdir -p ./artifacts/apicompat/
+source ./evergreen/packages.sh
+for package in ${PACKAGES[*]}; do
```
Comment (sanych-sun) [inline evergreen/validate-apicompat.sh:25]: Variable PACKAGE came from sourcing `./evergreen/packages.sh`. I've just want to make it impossible to forget to add a new package to validation script while adding it to publishing.

File: evergreen/evergreen.yml [hunk 2]
```diff
@@ -1833,6 +1833,21 @@ tasks:
         - func: build-packages
         - func: upload-packages
 
+    - name: validate-apicompat
+      commands:
+        - func: install-dotnet
+          vars:
+            DOTNET_SDK_VERSION: 6.0
```
Comment (papafe) [inline evergreen/evergreen.yml:1840]: How come we .net 6 here instead of 8?

**Review by papafe — COMMENTED** [review]
The script makes sense to me! Not much experience with evergreen though

File: evergreen/validate-apicompat.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# Environment variables used as input:
+# PACKAGE_VERSION
+
+if [ -z "$PACKAGE_VERSION" ]; then
+  PACKAGE_VERSION=$(bash ./evergreen/get-version.sh)
+  echo Calculated PACKAGE_VERSION value: "$PACKAGE_VERSION"
+fi
+
+BASELINE_VERSION="${PACKAGE_VERSION%%.*}.0.0"
+if [ "$PACKAGE_VERSION" == "$BASELINE_VERSION" ]; then
```
Comment (BorisDog) [inline evergreen/validate-apicompat.sh:13]: +1

**Review by BorisDog — APPROVED** [review]
LGTM!

File: evergreen/evergreen.yml (hunk 2 above)
Comment (sanych-sun) [reply evergreen/evergreen.yml:1840]: We need net6.0 to run the apicompat tool.

**Pull request merged**